replica/diff_disk.go

```diff
@@ -62,7 +62,7 @@ func (d *diffDisk) RemoveIndex(index int) error {
 	}
 
 	for i := 0; i < len(d.location); i++ {
-		if d.location[i] > uint16(index) {
+		if d.location[i] >= uint16(index) {
```

Comment:
I don't think that this is correct. Assuming indexes are from head to base for 0 to n and 'i'th index need to be removed.
i'th data is copied to i+1, i.e., i'th parent. And, now, it becomes i'th one. It consists of old i'th's file content and also 'i+1''s content. So, i'th index will not be change.

Reply:
Indexes are from base to head, head index is the largest, base index is 0

Reply:
ok.. can you fix the comments according to new logic and add few more comments

Reply:
can you verify that it leads to corruption without this change and testcase catches it?

Reply:
can I add this is next PRs